src/from_str.rs

```diff
@@ -41,7 +52,81 @@ pub fn expand(input: &DeriveInput, trait_name: &'static str) -> Result<TokenStre
                 Ok(#body)
             }
         }
-    })
+    }
+}
+
+fn enum_from(
+    input: &DeriveInput,
+    state: State,
+    trait_name: &'static str,
+) -> TokenStream {
+    let mut variants_caseinsensitive = HashMap::default();
+    for variant_state in state.enabled_variant_data().variant_states {
+        let variant = variant_state.variant.unwrap();
+        if variant.fields != Fields::Unit {
+            panic!("Only enums with no fields can derive({})", trait_name)
+        }
+
+        variants_caseinsensitive
+            .entry(variant.ident.to_string().to_lowercase())
+            .or_insert_with(Vec::new)
+            .push(variant.ident.clone());
+    }
+
+    let input_type = &input.ident;
+    let visibility = &input.vis;
+
+    let err_name = format_ident!("Parse{}Error", input_type);
+    let err_message =
+        format!("invalid {}", input_type.to_string().to_case(Case::Lower));
+
+    let mut cases = vec![];
+
+    // if a case insensitve match is unique match do that
+    // otherwise do a case sensitive match
+    for (ref canonical, ref variants) in variants_caseinsensitive {
+        if variants.len() == 1 {
+            let variant = &variants[0];
+            cases.push(quote! {
+                #canonical => #input_type::#variant,
+            })
+        } else {
+            for variant in variants {
+                let variant_str = variant.to_string();
+                cases.push(quote! {
+                    #canonical if(src == #variant_str) => #input_type::#variant,
+                })
+            }
+        }
+    }
+
+    let trait_path = state.trait_path;
+
+    quote! {
+
+        #[derive(Debug, Clone, PartialEq, Eq)]
+        #visibility struct #err_name;
+
+        impl std::fmt::Display for #err_name {
+            fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                fmt.write_str(#err_message)
+            }
+        }
+
+        impl std::error::Error for #err_name {}
+
+        impl #trait_path for #input_type
+        {
+            type Err = #err_name;
+            #[inline]
+            fn from_str(src: &str) -> ::core::result::Result<Self, Self::Err> {
+                Ok(match src.to_lowercase().as_str() {
```

Comment:
I considered adding `.replace(&[' ', '_', '-'], "")` here so that it would mach against kebab-case and SNAKE_CASE too.